### PR TITLE
Fix qesapdeployment unit testing

### PR DESCRIPTION
**Pull request by mpagot**
Remove previously added hot fix
Improve assert messages
Improve test isolation

- Verification run: github CI/CD

### CODE REVIEW

File: t/09_qesapdeployment.t [hunk 1]
```diff
@@ -8,41 +8,57 @@ use List::Util qw(any);
...
+    });
+
     my $inventory_path = qesap_get_inventory('nemo');
     note('inventory_path --> ' . $inventory_path);
-    is $inventory_path, '/root/qe-sap-deployment/terraform/nemo/inventory.yaml', "inventory_path:$inventory_path is the expected one";
+    is $inventory_path, '/BRUCE/terraform/nemo/inventory.yaml', "inventory_path:$inventory_path is the expected one";
 };
 
-subtest '[qesap_create_folder_tree] default' => sub {
+subtest '[qesap_create_folder_tree/qesap_get_file_paths] default' => sub {
```
Comment (alvarocarvajald) [inline t/09_qesapdeployment.t:38]: I propose removing `/qesap_get_file_paths` from the name of the subtest, and add a subtest for `qesap_get_file_paths` itself.

Could be something like:
```
subtest '[qesap_get_file_paths]' => sub {
    $qesap->redefine(data_url => sub { return 'url://qesap_conf_src/'; });    # needed by qesap_get_file_paths
    my %paths = qesap_get_file_paths();
    set_var('QESAP_CONFIG_FILE', 'qesap_file.config');
    my %expected_paths = (
        qesap_conf_filename => 'qesap_file.config',
        deployment_dir => '/root/qe-sap-deployment',
        terraform_dir => '/root/qe-sap-deployment/terraform',
        qesap_conf_trgt => '/root/qe-sap-deployment/scripts/qesap/qesap_file.config',
        qesap_conf_src => 'url://qesap_conf_src/qesap_file.config'
    );
    set_var('QESAP_CONFIG_FILE', undef);
    is_deeply \%paths, \%expected_paths, 'qesap_get_file_paths: default paths OK';
};
```

File: t/09_qesapdeployment.t [hunk 2]
```diff
@@ -8,41 +8,57 @@ use List::Util qw(any);
...
+
     qesap_create_folder_tree();
+
     note("\n  -->  " . join("\n  -->  ", @calls));
-    is $calls[0], 'mkdir -p /root/qe-sap-deployment';
+    is $calls[0], 'mkdir -p /root/qe-sap-deployment', "Default deploy location is /root/qe-sap-deployment";
 };
 
-subtest '[qesap_create_folder_tree] user specified deployment_dir' => sub {
+subtest '[qesap_create_folder_tree/qesap_get_file_paths] user specified deployment_dir' => sub {
```
Comment (alvarocarvajald) [inline t/09_qesapdeployment.t:50]: Same comment here regarding `/qesap_get_file_paths`.

**Review by alvarocarvajald — APPROVED** [review]
Pretty cool PR. Definitely like this more than dropping coverage.

I have 2 suggestions which is in reality only one, which should also help to increase test coverage, otherwise LGTM.

File: t/09_qesapdeployment.t (hunk 1 above)
Comment (mpagot) [reply t/09_qesapdeployment.t:38]: **qesap_get_file_paths** is not exported, is private. Your proposed test fails like:

```
# Subtest: [qesap_get_file_paths]
    1..0
    # No tests run!
not ok 3 - No tests run for subtest "[qesap_get_file_paths]"

#   Failed test 'No tests run for subtest "[qesap_get_file_paths]"'
#   at t/09_qesapdeployment.t line 54.
Undefined subroutine &main::qesap_get_file_paths called at t/09_qesapdeployment.t line 42.
        main::__ANON__() called at /usr/lib/perl5/5.36.0/Test/Builder.pm line 374
        eval {...} called at /usr/lib/perl5/5.36.0/Test/Builder.pm line 374
        Test::Builder::subtest(Test::Builder=HASH(0x55fe0e0a0428), "[qesap_get_file_paths]", CODE(0x55fe11860560)) called at /usr/lib/perl5/5.36.0/Test/More.pm line 809
        Test::More::subtest("[qesap_get_file_paths]", CODE(0x55fe11860560)) called at t/09_qesapdeployment.t line 54
 at /usr/lib/perl5/5.36.0/Test/Builder.pm line 418.
        Test::Builder::subtest(Test::Builder=HASH(0x55fe0e0a0428), "[qesap_get_file_paths]", CODE(0x55fe11860560)) called at /usr/lib/perl5/5.36.0/Test/More.pm line 809
        Test::More::subtest("[qesap_get_file_paths]", CODE(0x55fe11860560)) called at t/09_qesapdeployment.t line 54
ok 4 - no (unexpected) warnings (via END block)
# Tests were run but no plan was declared and done_testing() was not seen.
# Looks like your test exited with 255 just after 4.
```

Comment (alvarocarvajald) [reply t/09_qesapdeployment.t:38]: Oh, didn't realize it was private. I suppose if it's private, we don't need to test it directly. Disregard my 2 comments please.

**Review by alvarocarvajald — APPROVED** [review]
LGTM

**Pull request merged**